initfs/scripts/moslo-build.sh

```diff
@@ -58,6 +58,32 @@ debug()
         fi
 }
 
+LIB_PATHS="/usr/lib /lib"
+
+# Function to find _direct_ lib dependencies for executable using objdump
+objdump_find_lib_path()
+{
+        local FOUND_LIBS=""
+        local LIB_LIST=""
+        local FOUND_LIB=""
+
+        if test -z $1; then
+                echo "find_lib_path: Error: Empty executable name!"
+                exit 1
+        fi
+        LIB_LIST=$(objdump -p "$1" | grep NEEDED | sed -e 's/\NEEDED//g')
+        for lib in $LIB_LIST; do
+                for path in $LIB_PATHS; do
+                        FOUND_LIB=$(find $path/ | grep $lib)
```

Comment:
Fun.. It turns out the way I wrote was actually the thing that made it work by accident. It turns out objdump -p | grep NEEDED points to library symlinks only, and by grepping like I did, I get both the symlink and the actual library listed. Unfortunately it seems the order in which the library and symlink get in from "find" is random, and basename will not always match, even though the libraries are in fact correct. (comparison between symlink and library name is false, i.e libblkid.so.1 vs. libblkid.so.1.1.0).

I will just add a comment there.